### DISCUSSION TITLE
[Dagger/Nullable] issue when using generic type in subcomponent factory

### DISCUSSION THREAD

**Discussion by kevinguitar** [ZacSweers/anvil#65] [Q&A] [answered]
Hi Zac,

We've migrated successfully from Anvil to your fork (KSP only), there was one change required and it looks like a bug of `MergedComponent`, could you please advise 🙏 

Essentially, subcomponent factories don't allow generic without explicitly extending Any, although the type is nonnull in the use case.
```kotlin
interface AnvilAndroidInjector<T> {
    val injector: MembersInjector<T>

    fun inject(target: T) {
        injector.injectMembers(target)
    }

    interface Factory<T> {
        fun create(@BindsInstance instance: T): AnvilAndroidInjector<T>
    }
}


@MergeSubcomponent(scope = MainActivity::class)
interface MainActivitySubcomponent : AnvilAndroidInjector<MainActivity> {
    @MergeSubcomponent.Factory
    interface Factory : AnvilAndroidInjector.Factory<MainActivity>
}
```

I created a small project to replicate our DI mechanism with minimal code, and prepared a PR to demo the issue we're facing, hope it will help. Thank you!
https://github.com/kevinguitar/anvil-issue-replicate/pull/1

**Comment by ZacSweers** [comment]
This appears to just be coming from dagger itself. Is the request to duplicate this check in Anvil-KSP too?

**Reply by kevinguitar** [reply, marked as answer]
> Is the request to duplicate this check in Anvil-KSP too?

No, I'm just wondering why it complains about passing null to the factory method while the input isn't nullable. I guess it's required for dagger KSP now, thanks for the reply!
